### PR TITLE
New ThemeCard component

### PR DESCRIPTION
**Pull request by taipeicoder**
Related to #75457

## Proposed Changes

This PR creates the component `<ThemeCard />`, and updates the Theme Showcase to use this component to render theme cards. For this PR, `<ThemeCard />` still depend of several components to be passed as props (e.g.: banner, badge, or options). We can work on moving those components to be rendered within `<ThemeCard />` in future PRs. 

Screenshot 2023-04-10 at 12 14 06 PM

After this PR, it would be good to remove unused CSS code as a result of this change.

## Testing Instructions

* Head to the Theme Showcase.
* Ensure that the theme card UI and its functionality works the same as the one in production.
* Please also test the logged-out Theme Showcase.
* And also test Atomic sites.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/components/theme/index.jsx [hunk 1]
```diff
@@ -162,12 +141,6 @@ export class Theme extends Component {
 		this.props.onStyleVariationClick?.( this.props.theme.id, this.props.index, variation );
 	};
 
-	isBeginnerTheme() {
```
Comment (taipeicoder) [inline client/components/theme/index.jsx:165]: We currently do not have any themes that use this functionality, so I'm leaving this implementation out for the meantime.

File: packages/design-picker/src/components/theme-card/index.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+		const themeClasses = classnames( 'theme-card', {
+			'theme-card--is-active': isActive,
+			'theme-card--is-actionable': isActionable,
+		} );
+		const themeInfoClasses = classnames( 'theme-card__info', {
+			'theme-card__info--has-style-variations': styleVariations.length > 0,
+		} );
+
+		return (
+			<Card className={ themeClasses } onClick={ onClick } data-e2e-theme={ e2eName }>
```
Comment (taipeicoder) [inline packages/design-picker/src/components/theme-card/index.tsx:65]: There's an opportunity to break these elements into separate components as ThemeCard gets more complex.

**Review by okmttdhr — APPROVED** [review]
Worked as expected ✅

>After this PR, it would be good to remove unused CSS code as a result of this change.

👍

**Review by worldomonation — APPROVED** [review]
Approved from KitKat for changes relating to `packages/calypso-e2e`.

**Pull request merged**